sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptor/LogOptions.java

```diff
@@ -146,17 +159,26 @@ public String getApplicationId() {
      */
     public LogOptions setApplicationId(final String applicationId) {
         if (!CoreUtils.isNullOrEmpty(applicationId)) {
+            RuntimeException exception;
+
             if (applicationId.length() > MAX_APPLICATION_ID_LENGTH) {
-                throw logger
-                    .logExceptionAsError(new IllegalArgumentException("'applicationId' length cannot be greater than "
-                        + MAX_APPLICATION_ID_LENGTH));
+                exception = new IllegalArgumentException("'applicationId' length cannot be greater than "
+                    + MAX_APPLICATION_ID_LENGTH);
+
+                logger.error("Error found when setting Application ID", exception);
+
+                throw exception;
             } else if (applicationId.contains(" ")) {
-                throw logger
-                    .logExceptionAsError(new IllegalArgumentException("'applicationId' must not contain a space."));
+                exception = new IllegalArgumentException("'applicationId' must not contain a space.");
```

Comment:
I'm trying to think of a better way to handle the assertions about the format of the application ID. Suggestions?

Reply:
Ignore this comment (or at least apply it to the appropriate place in UserAgentInterceptor) if we remove this from LogOptions

Reply:
Aside from the User-Agent string, I'm not sure what we would want the Application ID for and how bad could it be if there is an error in the value passed on by customers. I'll ask the Java folks.